i2c-flash.c

```diff
@@ -22,12 +22,20 @@ ISR(TWI_vect) {
 
 	uint8_t offset;
 
-	switch (TWSR) {
+	switch (TW_STATUS) {
+	case TW_BUS_ERROR:
+		TWCR = _BV(TWEN) | _BV(TWIE) | _BV(TWEA) | _BV(TWINT) | _BV(TWSTO);
```

Comment:
I think this case should be written as `TWCR |= _BV(TWSTO) | _BV(TWINT)` followed by an early return. There's no point spinning on the `TWSTO` clear when we could just sleep and wait for the next interrupt instead. And the early return stops us from clearing the interrupt twice, which is probably harmless but isn't what the datasheet instructs.

Reply:
> I think this case should be written as `TWCR |= _BV(TWSTO) | _BV(TWINT)` followed by an early return. There's no point spinning on the `TWSTO` clear when we could just sleep and wait for the next interrupt instead. And the early return stops us from clearing the interrupt twice, which is probably harmless but isn't what the datasheet instructs.

Again: Read-modify-write == bad juju